src/effects/backends/effectsbackendmanager.cpp

```diff
@@ -0,0 +1,115 @@
+#include "effects/backends/effectsbackendmanager.h"
+
+#include "control/controlobject.h"
+#include "effects/backends/builtin/builtinbackend.h"
+#include "effects/backends/effectprocessor.h"
+#ifdef __LILV__
+#include "effects/backends/lv2/lv2backend.h"
+#endif
+#include "effects/presets/effectpreset.h"
+
+EffectsBackendManager::EffectsBackendManager() {
+    m_pNumEffectsAvailable = std::make_unique<ControlObject>(
+            ConfigKey("[Master]", "num_effectsavailable"));
+    m_pNumEffectsAvailable->setReadOnly();
+
+    addBackend(EffectsBackendPointer(new BuiltInBackend()));
+#ifdef __LILV__
+    addBackend(EffectsBackendPointer(new LV2Backend()));
+#endif
+}
+
+void EffectsBackendManager::addBackend(EffectsBackendPointer pBackend) {
+    VERIFY_OR_DEBUG_ASSERT(pBackend) {
+        return;
+    }
+
+    m_effectsBackends.insert(pBackend->getType(), pBackend);
+
+    QList<QString> backendEffects = pBackend->getEffectIds();
+    for (const QString& effectId : backendEffects) {
+        m_manifests.append(pBackend->getManifest(effectId));
+    }
+
+    m_pNumEffectsAvailable->forceSet(m_manifests.size());
+
+    std::sort(m_manifests.begin(),
+            m_manifests.end(),
+            EffectManifest::alphabetize);
+}
+
+const QList<EffectManifestPointer> EffectsBackendManager::getManifestsForBackend(
+        EffectBackendType backendType) const {
+    auto pBackend = m_effectsBackends.value(backendType);
+    VERIFY_OR_DEBUG_ASSERT(pBackend) {
+        return QList<EffectManifestPointer>();
+    }
+    auto list = pBackend->getManifests();
+    std::sort(list.begin(), list.end(), EffectManifest::alphabetize);
+    return list;
+}
+
+EffectManifestPointer EffectsBackendManager::getManifestFromUniqueId(
+        const QString& uid) const {
+    if (kEffectDebugOutput) {
+        //qDebug() << "EffectsBackendManager::getManifestFromUniqueId" << uid;
+    }
+    if (uid.isEmpty()) {
+        // Do not DEBUG_ASSERT, this may be a valid request for a nullptr to
+        // unload an effect.
+        return EffectManifestPointer();
+    }
+    int delimiterIndex = uid.lastIndexOf(" ");
+    EffectBackendType backendType =
+            EffectManifest::backendTypeFromString(uid.mid(delimiterIndex + 1));
+    VERIFY_OR_DEBUG_ASSERT(backendType != EffectBackendType::Unknown) {
```

Comment:
I'm hitting this debug assertion when starting Mixxx.
```
DEBUG ASSERT: "backendType != EffectBackendType::Unknown" in function EffectManifestPointer EffectsBackendManager::getManifestFromUniqueId(const QString&) const at /home/jan/Projects/mixxx/src/effects/backends/effectsbackendmanager.cpp:65
```

Effects do not work at all for me, probably due to the unknown backend.

Reply:
As explained in the comment below this, this debug assertion happens once when upgrading but should not happen again.

Reply:
Whoops, didn't see that.

Reply:
Still, the effect sections did not work at all (i.e. did not make an audible difference).

Reply:
Do EQs and QuickEffects work?

Reply:
Also can you try with a new directory for `--settingsPath`?

Reply:
Sorry, I had accidently turned down the dry-wet knob :hand_over_mouth: the effects do work.

Reply:
I keep hitting this debug assertion every time I test this branch, even though I tested it before with the same settings directory. Can we somehow detect the upgrade case and not debug assert in that case (only if it's not too much work)?